File: lib/modules/toolbox/models/toolbox_models.dart
/// 工具箱模块 - 单位转换配置模型
///
/// 用于定义单位转换的显示名称、目标单位以及转换逻辑。
class UnitConversionOption {
  /// 国际化显示的标签 Key
  final String labelKey;

  /// 转换结果的单位后缀 (如 'hPa', 'inHg')
  final String resultUnit;

  /// 具体的转换计算函数
  final double Function(double) converter;

  const UnitConversionOption({
    required this.labelKey,
    required this.resultUnit,
    required this.converter,
  });
}

/// 工具箱模块 - 航空术语模型
///
/// 用于定义航空术语的缩写、全称、中文名称及详细描述。
class AviationTerm {
  /// 术语缩写 (如 'V1', 'ILS')
  final String abbreviation;

  /// 术语英文全称
  final String fullName;

  /// 术语中文译名
  final String chineseName;

  /// 术语的详细解释 (可选)
  final String? description;

  const AviationTerm({
    required this.abbreviation,
    required this.fullName,
    required this.chineseName,
    this.description,
  });

  /// 获取用于搜索显示的复合字符串
  String get displayValue => '$chineseName ($fullName)';
}

File: src/modules/toolbox/models/toolbox-models.ts
/**
 * 工具箱数据模型
 * 对应 Flutter 版 `modules/toolbox/models/toolbox_models.dart`
 */

/** 单位转换配置项 */
export interface UnitConversionOption {
  /** 国际化显示的标签 key */
  readonly labelKey: string;
  /** 转换结果的单位后缀（如 'hPa' / 'inHg'） */
  readonly resultUnit: string;
  /** 具体的转换计算函数 */
  readonly converter: (value: number) => number;
}

/** 航空术语 */
export interface AviationTerm {
  /** 术语缩写（如 'V1' / 'ILS'） */
  readonly abbreviation: string;
  /** 英文全称 */
  readonly fullName: string;
  /** 中文译名 */
  readonly chineseName: string;
  /** 详细解释 */
  readonly description?: string;
}

/** 用于搜索展示的复合字符串 */
export function termDisplayValue(term: AviationTerm): string {
  return `${term.chineseName} (${term.fullName})`;
}

/** 工具箱分区（对应 AppBar 侧边二级菜单的 6 个入口） */
export type ToolboxSection =
  | 'unitConversion'
  | 'termTranslation'
  | 'flightCalculators'
  | 'weatherDecode'
  | 'performanceTools'
  | 'opsTools';

export const TOOLBOX_SECTIONS: ToolboxSection[] = [
  'unitConversion',
  'termTranslation',
  'flightCalculators',
  'weatherDecode',
  'performanceTools',
  'opsTools',
];
